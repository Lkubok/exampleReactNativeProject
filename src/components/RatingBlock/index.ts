import Component from './RatingBlock';

export default Component;
