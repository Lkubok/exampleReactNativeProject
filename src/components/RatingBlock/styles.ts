import { StyleSheet } from 'react-native';

import { fontNames, variables } from 'res';

export default StyleSheet.create({
  container: {},
  containerWrapper: {
    flexGrow: 1,
    marginBottom: variables.spacing.base,
  },
  emptyBox: {
    height: 10,
  },
  headerStyle: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: variables.spacing.small,
  },
  icon: {
    height: 30,
    marginRight: variables.spacing.min,
    width: 30,
  },
  link: {
    color: variables.colors.accentRed,
    fontFamily: fontNames.SFCompactTextSemibold,
    fontSize: variables.fontSize.medium,
  },
  messageStyle: {
    color: variables.colors.black,
    fontSize: variables.fontSize.base,
    lineHeight: variables.lineHeight.base,
  },
  ratingStars: {
    flexDirection: 'row',
    height: 30,
    marginBottom: variables.spacing.big,
    marginTop: variables.spacing.small,
  },
});
