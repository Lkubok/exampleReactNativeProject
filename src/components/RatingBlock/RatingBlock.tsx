import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Text, View } from 'react-native';
import { times } from 'lodash';

import { variables } from 'res';
import { Typography } from 'components';
import { useTranslations } from 'hooks';

import IconRatingEmpty from 'assets/iconRatingEmpty.svg';

import styles from './styles';
interface Props {
  setRating: (rating: number) => void;
  rating: number;
  onDismiss: () => void;
}

const RatingBlock: FC<Props> = ({ setRating, rating, onDismiss }) => {
  const { i18n } = useTranslations();

  const ratingContent = times(5, (index) => (
    <TouchableOpacity
      onPress={() => setRating(index + 1)}
      hitSlop={{ bottom: 12, left: 12, right: 12, top: 12 }}
      key={index}>
      <IconRatingEmpty
        style={styles.icon}
        fill={rating > index ? variables.colors.orange_primary : variables.colors.white}
      />
    </TouchableOpacity>
  ));

  return (
    <View style={styles.containerWrapper}>
      <View style={styles.container}>
        <View style={styles.headerStyle}>
          <Typography type="subheading" text={i18n.t('shop_rating')} />
          <TouchableOpacity onPress={onDismiss} hitSlop={{ bottom: 12, left: 12, right: 12, top: 12 }}>
            <Text style={styles.link}>{i18n.t('messages_rating_dismiss')}</Text>
          </TouchableOpacity>
        </View>
        <Text style={styles.messageStyle}>{i18n.t('messages_rating_hello')}</Text>
        <Text style={styles.messageStyle}>{i18n.t('messages_rating_text')}</Text>
        <View style={styles.emptyBox} />
        <Text style={styles.messageStyle}>{i18n.t('messages_rating_satisfaction')}</Text>
        <View style={styles.ratingStars}>{ratingContent}</View>
      </View>
    </View>
  );
};

export default RatingBlock;
