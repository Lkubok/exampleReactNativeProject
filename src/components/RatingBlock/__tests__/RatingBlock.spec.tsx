import React from 'react';
import { act, customRender } from 'utils/testUtils';

import RatingBlock from '../RatingBlock';

jest.useFakeTimers();

describe('RatingBlock component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      autoCapitalize: 'words',
      blocked: false,
      error: 'testError',
      initialFocus: true,
      isPasswordField: true,
      keyboardType: 'number-pad',
      label: 'test label',
      onBlur: () => {},
      onChange: () => {},
      onChangeText: () => {},
      onFocus: () => {},
      onSubmitEditing: () => {},
      placeholder: 'test placeholder',
      secureTextEntry: true,
      shakeCounter: 5,
      useNativeDriver: true,
      value: 'test value',
    };
  });

  it('should match snapshot when 1 line', () => {
    const snapshot = customRender(<RatingBlock {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
