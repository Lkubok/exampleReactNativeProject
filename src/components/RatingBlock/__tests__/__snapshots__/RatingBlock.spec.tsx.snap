// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingBlock component should match snapshot when 1 line 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexGrow": 1,
        "marginBottom": 12,
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 8,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontFamily": "SFCompactText-Semibold",
                "fontSize": 18,
                "letterSpacing": -0.36,
              }
            }
          >
            Shop rating
          </Text>
        </View>
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "[7m[38;2;234;51;54m#ea3336[39m[27m",
                  "fontFamily": "SFCompactText-Semibold",
                  "fontSize": 18,
                }
              }
            >
              Dismiss
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontSize": 17,
            "lineHeight": 21,
          }
        }
      >
        Good morning,
      </Text>
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontSize": 17,
            "lineHeight": 21,
          }
        }
      >
        If you have already closed a deal, please write a review. Your feedback will not only help improve our services, but will also serve others. Thank you in advance
      </Text>
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontSize": 17,
            "lineHeight": 21,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontSize": 17,
            "lineHeight": 21,
          }
        }
      >
        How satisfied were you with the deal?
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 30,
            "marginBottom": 24,
            "marginTop": 8,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <SvgMock
              fill="#ffffff"
              style={
                Object {
                  "height": 30,
                  "marginRight": 4,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <SvgMock
              fill="#ffffff"
              style={
                Object {
                  "height": 30,
                  "marginRight": 4,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <SvgMock
              fill="#ffffff"
              style={
                Object {
                  "height": 30,
                  "marginRight": 4,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <SvgMock
              fill="#ffffff"
              style={
                Object {
                  "height": 30,
                  "marginRight": 4,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
        <RNGestureHandlerButton
          collapsable={false}
          hitSlop={
            Object {
              "bottom": 12,
              "left": 12,
              "right": 12,
              "top": 12,
            }
          }
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
        >
          <View
            accessible={true}
            hitSlop={
              Object {
                "bottom": 12,
                "left": 12,
                "right": 12,
                "top": 12,
              }
            }
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <SvgMock
              fill="#ffffff"
              style={
                Object {
                  "height": 30,
                  "marginRight": 4,
                  "width": 30,
                }
              }
            />
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
